### ISSUE TITLE
Remove required dependency on JSON, store binary data

### ISSUE THREAD

**Issue by wohali** [apache/couchdb#1537]
@chewbranca 's idea - store binary natively without a JSON interface.

Not sure how this would work, maybe the same way attachments do, but without multipart ugliness?

**Comment by mikerhodes** [comment]
@wohali Another thought I had in this vein is storing JSON-as-binary, as described at https://github.com/cockroachdb/cockroach/blob/master/docs/RFCS/20171005_jsonb_encoding.md for a faster processing pipeline. It sounds like this idea is different though?

**Comment by wohali** [comment]
@mikerhodes Yup, that's different, this is about accepting binary data on the HTTP API (though I know that we previously said we'd not do BSON)

**Comment by mikerhodes** [comment]
jsonb (Postgres-originate) is different to BSON (Mongo), unsure whether they have different trade-offs that change the balance of that discussion. jsonb seems to have a nice format to enable faster queries which could benefit Mango where it loads up documents and matches them against a selector.

**Comment by wohali** [comment]
@mikerhodes but this sounds like it's more about an internal storage format, not for external communication? I might be confused. If what you're talking about is over-the-wire comms with a client for transmitting non-JSON data, then this issue applies. If you're talking about just a way to accelerate transmitting JSON data, file a new ticket please. :)

**Comment by mikerhodes** [comment]
My bad :) I'll leave it for now.
